src/main/java/mate/academy/DAO/BookDaoImpl.java

```diff
@@ -0,0 +1,103 @@
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.executeUpdate();
+
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getLong(1));
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error creating book: " + book, e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(FIND_BY_ID_QUERY)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(parseBook(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error retrieving book with id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(FIND_ALL_QUERY)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<Book> books = new ArrayList<>();
+            while (resultSet.next()) {
+                books.add(parseBook(resultSet));
+            }
+            return books;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error retrieving all books", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(UPDATE_BOOK_QUERY)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.setLong(3, book.getId());
+            preparedStatement.executeUpdate();
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error updating book: " + book, e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(DELETE_BY_ID_QUERY)) {
+            preparedStatement.setLong(1, id);
+            int rowsDeleted = preparedStatement.executeUpdate();
+            return rowsDeleted > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error deleting book with id: " + id, e);
+        }
+    }
+
+    private Book parseBook(ResultSet resultSet) throws SQLException {
```

Comment:
To avoid initializing the `Book` object inconsistently, use either setters or the constructor, but not both. Here, you are using the constructor which is good practice.